src/main/java/org/javarosa/core/util/ArrayUtilities.java

```diff
@@ -33,6 +35,20 @@ public static boolean arraysEqual(byte[] array1, byte[] array2) {
     }
 
     /**
+     * Reverses an Array
+     */
+    public static <T> T[] reverse(Object[] myArray, Class<? extends T[]> tClass) {
```

Comment:
If you switch to using a `char[]` you could use Apache Commons `ArrayUtils.reverse` https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/ArrayUtils.html#reverse-char:A-